internal/server/backgroundjobs_test.go

```diff
@@ -0,0 +1,90 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+	"gotest.tools/v3/assert"
+
+	"github.com/infrahq/infra/internal"
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/models"
+)
+
+func TestJobWrapper(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	db := setupDB(t)
+
+	var signal struct{} // used to signal goroutines to continue
+	chReady := make(chan struct{})
+	chRun := make(chan struct{})
+
+	runOnce := func() {
+		chRun <- signal
+		<-chReady
+	}
+
+	var runID int
+	job := func(ctx context.Context, tx *data.Transaction) error {
+		chReady <- signal
+		<-chRun
```

Comment:
This allows us to test the loop one iteration at a time.

The background goroutine will signal when it's ready, and wait for a signal from the main test goroutine before running.

Each test case will call `runOnce` which tells the background goroutine to run, and wait for the loop to get back to `chReady` before making assertions about the run.